corehq/util/test_utils.py

```diff
@@ -195,3 +196,56 @@ def __exit__(self, exc_type, exc_val, exc_tb):
             self.logger.removeHandler(handler)
         for handler in self.original_handlers:
             self.logger.addHandler(handler)
+
+
+def generate_cases(argsets, cls=None):
```

Comment:
It would be possible to make this produce test names that are able to be selected and run individually. The trade-off is test failure readability, and so far I haven't had the need to change that. An even better solution would be the ability to write [nose generator tests](http://nose.readthedocs.org/en/latest/writing_tests.html#test-generators), but we can't do that until nose is our primary test runner.

It may be worth generalizing your approach if you find yourself using it often.